examples/factors/README.md

```diff
@@ -1,22 +1,203 @@
+You can confirm your work with `cargo run --release` -- the program still won't do anything, but it should compile and run successfully.
+You will get a warning about `from_slice` being unused; this function is needed when reading the journal, but the guest hasn't written to it yet, so the host can't read from it yet.
+
+## Step 4 (Guest): Multiply two values and commit their result
+
+Now it's time to start writing guest code.
+Open the main guest program file `methods/guest/src/main.rs`.
+In its final form, we'll tell the guest to read the values of `a` and `b` from the host and multiply them together.
+We'll then publicly commit their product to the `receipt` portion of the `journal`.
+
+Here is the complete guest program.
+We'll break this down step by step below:
+```rust
+pub fn main() {
+    // Load the first number from the host
+    let a: u64 = env::read();
+    // Load the second number from the host
+    let b: u64 = env::read();
+    // Verify that neither of them are 1 (i.e. nontrivial factors)
+    if a == 1 || b == 1 {
+        panic!("Trivial factors")
+    }
+    // Compute the product while being careful with integer overflow
+    let product = a.checked_mul(b).expect("Integer overflow");
+    env::commit(&product);
+}
+```
+### Load values from the host
+
+First, uncomment `use risc0_zkvm::guest::env;` at the top of the file (outside of the function body) to bring `env` into scope.
+
+Then, we use `env::read()` to load both numbers:
+
+```rust
+    let a: u64 = env::read();
+    let b: u64 = env::read();
+```
+### Confirm that factors are non-trivial
+
+Next, we'll add a line that panics if either chosen value is 1. This will leave us with a guest program that only completes if the product of `a` and `b` is genuinely composite. 
+```
+    // Verify that neither of them are 1 (i.e. nontrivial factors)
+    if a == 1 || b == 1 {
+        panic!("Trivial factors")
+    }
+```
+
+Now we can compute their product and `commit` it.
+Once committed to the `journal`, anyone with the receipt can read this value.
+
+Now you should be able to confirm your work again with `cargo run --release`.
+Once more, the program won't do anything, but it should run successfully and build with a single warning about `from_slice` being unused.
+
+## Step 5 (Host): Generate a receipt and read its journal contents
+
+For this step, we return to the main file for the host driver program at `factors/host/src/main.rs`, which currently has a placeholder comment asking to fill in with code for handling the receipt:
+
+```rust
+    // TODO: Implement code for transmitting or serializing the receipt for
+    // other parties to verify here
+```
+
+In a real-world scenario, we'd want to hand the receipt to someone else, but reading it ourselves will be a nice way to check our project is working as expected.
+So, let's extract the journal's contents by replacing the lines above with the following lines.
```

Comment:
again, I find "the lines above" to be a bit ambiguous -- I think language like "replacing the TODO in the previous code snippet" is preferable